Migrant/VersionTolerance/ArrayDescriptor.cs

```diff
@@ -0,0 +1,69 @@
+﻿// *******************************************************************
```

Comment:
Unfortunately, `ArrayDescriptor` is constructed from two parameters: `elementType` and `ranks` array. The first one is read from stream much earlier, so it is not possible to read whole descriptor at once. As a result I would stick with current solution.